llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -362,6 +364,34 @@ static bool expandClampIntrinsic(CallInst *Orig, Intrinsic::ID ClampIntrinsic) {
   return true;
 }
 
+static bool expandSaturateIntrinsic(CallInst *SaturateCall) {
```

Comment:
wait for @pow2clk  change merges then change this to return `Result` and remove the `replaceAllUsesWith` and `eraseFromParent` calls